test/functionality/_helper/shell.ts

```diff
@@ -453,19 +453,18 @@ export function assertSliced(
 	input: string,
 	criteria: SlicingCriteria,
 	expected: string,
-	userConfig?: Partial<TestConfigurationWithOutput> & { autoSelectIf?: AutoSelectPredicate, skipTreeSitter?: boolean, skipCompare?: boolean, cfgExcludeProperties?: readonly CfgProperty[], flowrConfig?: FlowrConfigOptions },
-	testCaseFailType?: TestCaseFailType,
-	getId: () => IdGenerator<NoInfo> = () => deterministicCountingIdGenerator(0)
+	userConfig?: Partial<TestConfigurationWithOutput> & { autoSelectIf?: AutoSelectPredicate, skipTreeSitter?: boolean, skipCompare?: boolean, cfgExcludeProperties?: readonly CfgProperty[], flowrConfig?: FlowrConfigOptions, testCaseFailType?: TestCaseFailType, getId?: () => IdGenerator<NoInfo> },
```

Comment:
Could we additionally take all the stuff from the intersection (`{...}`) and move it into an interface so that we can comment on the meaning of each individual property? Ideally the interface does not use `|undefined` and then we wrap it within a `Partial` similar to the `TestConfigurationWithOutput`